internal/qmp-gen/main.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2016 The go-qemu Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gen
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"go/format"
+	"io/ioutil"
+)
+
+const specURL = `https://raw.githubusercontent.com/danderson/qemu/master/qapi-schema.json`
```

Comment:
The official spec used to have inconsistencies that made it hard to parse. The patch I submitted upstream has been published (minus one regression which they're fixing), so now it's fine to use upstream again. Fixed.